Source/WebKit/gtk/tutorial.md

```diff
@@ -0,0 +1,382 @@
+# Create a Web Browser in GTK with WebKit
+
+This is a tutorial for creating a simple web browser with some basic functionalities in GTK with WebKit in C language.
+
+GTK is a multi-platform toolkit for creating graphical user interfaces; it contains a complete set of widgets to be reused and combined together in development.
+
+## Prerequisites
+
+(The following examples are run on Ubuntu 20.04 LTS.)
+
+First the two required libraries need to be installed:
+
+```sh
+$ sudo apt-get install libgtk-3-dev
+$ sudo apt-get install libwebkit2gtk-4.0-dev
+```
+
+### Installing Meson build system with pip
+
+Meson is available in the [Python Package Index](https://pypi.python.org/pypi/meson/) and can be installed with `pip3 install --user meson`. This does not require any special privileges. This will install the package in `~/.local/`, so you will have to add `~/.local/bin` to your `PATH`.
+
+```sh
+$ sudo apt-get install python3 python3-pip python3-setuptools \
+                       python3-wheel ninja-build
+$ pip3 install --user meson
+```
+
+## Coding a simple browser
+
+In our `browse1.c` file, we add the following content to it. In the main function we create a GtkApplication called `app` and connect to it a callback function `on_app_activate`, in which a `web_view` GtkWidget loading a specified URI is created and placed in a window container.
+
+```c
+#include <webkit2/webkit2.h>
+
+static void
+on_app_activate(GtkApplication *app)
+{
+    GtkWidget *web_view = webkit_web_view_new();
+    webkit_web_view_load_uri(WEBKIT_WEB_VIEW(web_view), "http://ddg.gg");
+
+    GtkWidget* window = gtk_application_window_new(app);
+    // Adds the web_view widget to the window container.
+    gtk_container_add(GTK_CONTAINER(window), web_view);
+    gtk_window_set_default_size(GTK_WINDOW(window), 800, 600);
+
+    gtk_application_add_window(app, GTK_WINDOW(window));
+    // Recursively shows a widget, and any child widgets (if the widget is a container).
+    gtk_widget_show_all(window);
+}
+
+int
+main(int argc, char *argv[])
+{
+    g_autoptr(GtkApplication) app = gtk_application_new(NULL, G_APPLICATION_FLAGS_NONE);
+    gtk_window_set_default_icon_name("internet-web-browser");
+    g_signal_connect(app, "activate", G_CALLBACK(on_app_activate), NULL);
+    return g_application_run(G_APPLICATION(app), argc, argv);
+}
+```
+
+## Building and running the browser
+
+Then we create a Meson build description and put it in a file called `meson.build` in the directory. Its contents are the following:
```

Comment:
The example programs are only one `.c` file so I think we can do like [the GTK “getting started”](https://docs.gtk.org/gtk3/getting_started.html#hello-world-in-c) and mentioning a one-line command to compile the example:

```sh
cc -o browse1 browse1.c $(pkg-config webkit2gtk-4.1 --cflags --libs)
```

This way we can skip mentioing Meson and make the tutorial focus on the example programs. That should make it a bit easier to follow.